protobuf/src/main/java/io/grpc/protobuf/ProtoUtils.java

```diff
@@ -73,7 +74,12 @@ public T parse(InputStream stream) {
           }
         }
         try {
-          return parser.parseFrom(stream);
+          // Pre-create the CodedInputStream so that we can remove the size limit restriction
+          // when parsing.
+          CodedInputStream codedInput = CodedInputStream.newInstance(stream);
+          codedInput.setSizeLimit(Integer.MAX_VALUE);
```

Comment:
This is part of #832. Basically, part 1 is remove the limit from protobuf. Part 2 will be to enforce our own limit within gRPC itself (in a marshaller-agnostic way).